src/core/ext/client_config/lb_policy_factory.h

```diff
@@ -47,8 +47,24 @@ struct grpc_lb_policy_factory {
   const grpc_lb_policy_factory_vtable *vtable;
 };
 
+/** A resolved address alongside any LB related information associated with it.
+ * \a user_data, if not \a NULL, is opaque and meant to be consumed by the gRPC
+ * LB policy. Anywhere else, refer to the functions in \a
+ * grpc_lb_policy_user_data_vtable to operate with it */
+typedef struct grpc_lb_address {
+  grpc_resolved_address *resolved_address;
+  void *user_data;
```

Comment:
Please document the fact that not all LB policies support user data.

Also, please add a check to both grpclb.c and pick_first.c to ensure that this is set to NULL, so that callers don't accidentally try passing user data in to those policies.